doc/source/whatsnew/v0.18.1.txt

```diff
@@ -76,8 +76,7 @@ API changes
 
 
 
-
-
+- In ``read_csv``, the Python parser now raises an ``EmptyDataError`` in response to empty columns or header (:issue:`12506`)
```

Comment:
and is that an actual test that you had to change? I think the answer is no, so pls add that to test coverage. you have to be extraordinarily careful when attempting to change exceptions where we may not necessarily have tests coverage.